verifier/trustpolicy/blob.go

```diff
@@ -0,0 +1,144 @@
+	TrustStores []string `json:"trustStores"`
+
+	// TrustedIdentities this policy statement pins
+	TrustedIdentities []string `json:"trustedIdentities"`
+
+	// GlobalPolicy defines if policy statement is global or not
+	GlobalPolicy bool `json:"globalPolicy,omitempty"`
+}
+
+var supportedBlobPolicyVersions = []string{"1.0"}
+
+// LoadBlobDocument loads a trust policy document from a local file system
+func LoadBlobDocument() (*BlobDocument, error) {
+	var doc BlobDocument
+	err := getDocument(dir.PathBlobTrustPolicy, &doc)
+	return &doc, err
+}
+
+// Validate validates a policy document according to its version's rule set.
+// if any rule is violated, returns an error
+func (policyDoc *BlobDocument) Validate() error {
+	// sanity check
+	if policyDoc == nil {
+		return errors.New("blob trust policy document cannot be nil")
+	}
+
+	// Validate Version
+	if policyDoc.Version == "" {
+		return errors.New("blob trust policy has empty version, version must be specified")
+	}
+	if !slices.Contains(supportedBlobPolicyVersions, policyDoc.Version) {
+		return fmt.Errorf("blob trust policy document uses unsupported version %q", policyDoc.Version)
+	}
+
+	// Validate the policy according to 1.0 rules
+	if len(policyDoc.TrustPolicies) == 0 {
+		return errors.New("blob trust policy document can not have zero trust policy statements")
+	}
+
+	policyNames := set.New[string]()
+	var foundGlobalPolicy bool
+	for _, statement := range policyDoc.TrustPolicies {
+		// Verify unique policy statement names across the policy document
+		if policyNames.Contains(statement.Name) {
+			return fmt.Errorf("multiple blob trust policy statements use the same name %q, statement names must be unique", statement.Name)
+		}
+
+		if err := validatePolicyCore(statement.Name, statement.SignatureVerification, statement.TrustStores, statement.TrustedIdentities); err != nil {
+			return fmt.Errorf("blob trust policy: %w", err)
+		}
+
+		if statement.GlobalPolicy {
+			if foundGlobalPolicy {
+				return errors.New("multiple blob trust policy statements have globalPolicy set to true. Only one trust policy statement should be marked as global policy")
+			}
+			foundGlobalPolicy = true
+		}
+		policyNames.Add(statement.Name)
+	}
+
+	return nil
+}
+
+// GetApplicableTrustPolicy returns a pointer to the deep copied TrustPolicy for given policy name
+// see https://github.com/notaryproject/notaryproject/blob/v1.1.0/specs/trust-store-trust-policy.md#blob-trust-policy
+func (policyDoc *BlobDocument) GetApplicableTrustPolicy(policyName string) (*BlobTrustPolicy, error) {
+	if strings.TrimSpace(policyName) == "" {
+		return  nil, errors.New("policy name cannot be empty")
+	}
+	for _, policyStatement := range policyDoc.TrustPolicies {
+		// exact match
+		if policyStatement.Name == policyName {
+			return (&policyStatement).clone(), nil
+		}
+	}
+
+	return nil, fmt.Errorf("no applicable blob trust policy. Applicability for a given blob is determined by policy name")
```

Comment:
We should explicitly show which trust policy was searched and not found.
```suggestion
	return nil, fmt.Errorf("no applicable blob trust policy with name %q", policyName)
```